### PR TITLE
Kernel module

### PR DESCRIPTION
**Pull request by Waujito**
On low-level embedded devices, such as routers performance is extremly important. For example, my OpenWRT 1 Core router with userspace youtubeUnblock enabled slowdowns the Internet speed from 12 MiB/s to 2 MiB/s. And all performanse got lost on multiple context switches with kenel: two per each packet. So the solution _may_ be a module that deeply integrates with iptables, without userspace netfilter queue.

### CODE REVIEW

**Comment by Waujito** [comment]
Special thanks to https://github.com/drivenets/iptables_extensions repository with detailed explanations of iptables userspace interaction with the kernel module.
It helped me a lot!

File: iptk_YTUNBLOCK.c [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+	if (pktlen > AVAILABLE_MTU) {
+		pr_alert("The packet is too big!");
+		return -ENOMEM;
+#ifdef DEBUG
+		printf("Split packet!\n");
+#endif
+
+		__u32 buff1_size = pktlen;
+		__u32 buff2_size = pktlen;
+		__u8 *buff1 = kmalloc(pktlen, GFP_KERNEL);
```
Comment (klirichek) [inline iptk_YTUNBLOCK.c:81]: m.b. buff1_size instead of pktlen here?

**Comment by Waujito** [comment]
[3f1e8d3](https://github.com/Waujito/youtubeUnblock/pull/9/commits/3f1e8d3f013706a24f40e351fa07922afcd26ba1) is related to #32

**Comment by jelly8173** [comment]
Speaking about performance, how exactly did you measured the bottleneck?

On my _very powerfull_ 1 MIPS core 320Mhz sagem fast 2704 router i get 51.48Mbps upload speed with yt-unblk and 51.88Mbps without

**Comment by Waujito** [comment]
> Speaking about performance, how exactly did you measured the bottleneck?
> 
> On my _very powerfull_ 1 MIPS core 320Mhz sagem fast 2704 router i get 51.48Mbps **upload** speed with yt-unblk and 51.88Mbps without

If you're using ct packets counter everything is alright. If no - just cry. Jumps between kernel-space and user-space for each packet performance are awful.

Also a big role plays -20 priority but I don't want to see that my router is 100% on 100MBit/s

**Comment by jelly8173** [comment]
> If you're using ct packets counter everything is alright. If no - just cry. Jumps between kernel-space and user-space for each packet are awful.
> 
> Also a big role plays -20 priority but I don't want to see that my router is unusable and under 100% on 100MBit/s

Hey, I use ct packets counter and no speed issues at all. And now I wonder what happens without counter, maybe something will work different in long run, like HTTP/2 come back? What other benefits except loss of speed can i gain disabling it?

**Pull request merged**